File: app/src/main/java/com/esoxjem/movieguide/favorites/FavoritesModule.java
package com.esoxjem.movieguide.favorites;

import com.esoxjem.movieguide.AppModule;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * @author pulkitkumar
 */
@Module(includes = AppModule.class)
public class FavoritesModule
{
    @Provides
    @Singleton
    IFavoritesInteractor provideFavouritesInteractor(FavoritesStore store)
    {
        return new FavoritesInteractor(store);
    }
}

File: app/src/main/java/com/esoxjem/movieguide/movies/favorites/FavoritesModule.kt
package com.esoxjem.movieguide.movies.favorites

import com.esoxjem.movieguide.AppModule

import javax.inject.Singleton

import dagger.Module
import dagger.Provides

/**
 * @author pulkitkumar
 */
@Module(includes = [(AppModule::class)])
class FavoritesModule {
    @Provides
    @Singleton
    fun provideFavouritesInteractor(store: FavoritesStore): FavoritesInteractor {
        return FavoritesInteractorImpl(store)
    }
}
